### DISCUSSION TITLE
Add password support for files/folders

### DISCUSSION THREAD

**Discussion by rijkvanzanten** [directus/directus#3063] [Feature Requests]
Allows folders / files to be password protected, allowing the user to make a folder/file private

**Comment by benhaynes** [comment]
@rijkvanzanten — is this a new feature? Didn't we have private files (via the URL) in v8? Or is this based on permissions?

**Comment by rijkvanzanten** [comment]
This is a new feature. It's a way to make files "public" (eg you don't need an account), but still "private" (eg you need a password to download the file).

It's effectively a way to share files "privately publicly" to people outside of Directus

**Comment by rgieseke** [comment]
Any thoughts how could this be implemented? Would it be possible to build it with hooks?

**Reply by rijkvanzanten** [reply]
I think we could achieve this by relying on (parts of) the shares feature we already have. You can currently share a single item from a collection with password protection. If you we would allow sharing a file detail page, you could theoretically share the file password protected!

**Reply by rgieseke** [reply]
Where would this need to be allowed?
I looked through https://github.com/directus/directus/commit/dbf35a1736f8068a9bbc768cb5645d96c0777285 but didn't spot something yet.

**Reply by rijkvanzanten** [reply]
We'd have to add the share drawer detail to the files module, and then update the public share form to include a file preview component when the collection is directus_files 🤔 That might already be enough to support this!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
